clickstream-health-metrics/src/main/kotlin/clickstream/health/internal/DefaultCSHealthEventProcessor.kt

```diff
@@ -83,14 +84,16 @@ public class DefaultCSHealthEventProcessor(
 
     override fun onStart() {
         logger.debug { "DefaultCSHealthEventProcessor#onStart" }
-
-        scope?.cancel()
-        scope = null
+        if (scope.isActive) {
+            scope.cancel()
```

Comment:
there's a reason we set it as null. so that the old reference on the heap would be GCied

Reply:
unless we want to reuse the scope we can use cancelChildren() instead of cancel